spec/classes/prosody_spec.rb

```diff
@@ -13,8 +13,8 @@
         it { is_expected.to contain_class 'prosody::service' }
 
         it { is_expected.to contain_package('prosody').with(ensure: 'present') }
-        it { is_expected.to contain_file('/etc/prosody/conf.avail').with(ensure: 'directory') }
-        it { is_expected.to contain_file('/etc/prosody/conf.d').with(ensure: 'directory') }
+        it { is_expected.to contain_file("#{prosody::config_directory}/conf.a'ail").with(ensure: 'directory') }
+        it { is_expected.to contain_file("#{prosody::config_directory}/conf.d").with(ensure: 'directory') }
```

Comment:
`$prosody::config_directory` is something of the Puppet language that you don't really have access to here in Ruby code.

In this case, you can do something like:

```ruby
config_directory = case facts[:osfamily]
                   when 'Gentoo' then '/etc/foo'
                   else '/etc/bar'
                   end

it { is_expected.to contain_file("#{config_directory}/conf.d").with(ensure: 'directory') }
```